src/wallet/wallet_tx_builder.h

```diff
@@ -25,4 +28,272 @@ class ResolvedPayment : public RecipientMapping {
+        for (const ResolvedPayment& payment : payments) {
+            std::visit(match {
+                [&](const CKeyID& addr) {
+                    t_outputs_total += payment.amount;
+                    recipientPools.insert(OutputPool::Transparent);
+                },
+                [&](const CScriptID& addr) {
+                    t_outputs_total += payment.amount;
+                    recipientPools.insert(OutputPool::Transparent);
+                },
+                [&](const libzcash::SaplingPaymentAddress& addr) {
+                    sapling_outputs_total += payment.amount;
+                    recipientPools.insert(OutputPool::Sapling);
+                },
+                [&](const libzcash::OrchardRawAddress& addr) {
+                    orchard_outputs_total += payment.amount;
+                    recipientPools.insert(OutputPool::Orchard);
+                }
+            }, payment.address);
+        }
+    }
+
+    const std::set<OutputPool>& GetRecipientPools() const {
+        return recipientPools;
+    }
+
+    bool HasTransparentRecipient() const {
+        return recipientPools.count(OutputPool::Transparent) > 0;
+    }
+
+    bool HasSaplingRecipient() const {
+        return recipientPools.count(OutputPool::Sapling) > 0;
+    }
+
+    bool HasOrchardRecipient() const {
+        return recipientPools.count(OutputPool::Orchard) > 0;
+    }
+
+    const std::vector<ResolvedPayment>& GetResolvedPayments() const {
+        return payments;
+    }
+
+    CAmount GetTransparentBalance() const {
+        return t_outputs_total;
+    }
+
+    CAmount GetSaplingBalance() const {
+        return sapling_outputs_total;
+    }
+
+    CAmount GetOrchardBalance() const {
+        return orchard_outputs_total;
+    }
+
+    CAmount Total() const {
+        return
+            t_outputs_total +
+            sapling_outputs_total +
+            orchard_outputs_total;
+    }
+};
+
+typedef std::variant<
+    RecipientAddress,
+    SproutPaymentAddress> ChangeAddress;
+
+class TransactionEffects {
+private:
+    AccountId sendFromAccount;
+    uint32_t anchorConfirmations{1};
+    SpendableInputs spendable;
+    Payments payments;
+    ChangeAddress changeAddr;
+    CAmount fee{0};
+    uint256 internalOVK;
+    uint256 externalOVK;
+
+public:
+    TransactionEffects(
+        AccountId sendFromAccount,
+        uint32_t anchorConfirmations,
+        SpendableInputs spendable,
+        Payments payments,
+        ChangeAddress changeAddr,
+        CAmount fee,
+        uint256 internalOVK,
+        uint256 externalOVK) :
+            sendFromAccount(sendFromAccount),
+            anchorConfirmations(anchorConfirmations),
+            spendable(spendable),
+            payments(payments),
+            changeAddr(changeAddr),
+            fee(fee),
+            internalOVK(internalOVK),
+            externalOVK(externalOVK) {}
+
+    PrivacyPolicy GetRequiredPrivacyPolicy() const;
```

Comment:
Document that this returns the strongest privacy policy that is compatible with this transaction's effects.